backend/src/main/java/de/laura/project/service/TriviaService.java

```diff
@@ -14,26 +16,76 @@
 public class TriviaService {
 
     private final TriviaApiService triviaApiService;
+    private final TempTriviaQuestionDB tempTriviaQuestionDB;
+    private final AnswerRandomizerService answerRandomizerService;
+
+
+
 
     @Autowired
-    public TriviaService(TriviaApiService triviaApiService) {
+
+    public TriviaService(TriviaApiService triviaApiService, TempTriviaQuestionDB tempTriviaQuestionDB, AnswerRandomizerService answerRandomizerService) {
         this.triviaApiService = triviaApiService;
+        this.tempTriviaQuestionDB = tempTriviaQuestionDB;
+        this.answerRandomizerService = answerRandomizerService;
     }
 
-    public List<TriviaQuestionSet> getListDataPoints() {
 
-        List<TriviaApiData> triviaApiDataList = triviaApiService.getDataPoints().getTriviaApiData();
 
-        return triviaApiDataList.stream()
+
+    public List<TriviaQuestionSet> callQuestionList(int amount, int category, String difficulty) {
+
+        List<TriviaApiData> triviaApiDataList = triviaApiService.callQuestionList(amount, category, difficulty);
+
+        List<TriviaQuestionSet> triviaQuestionSetList = new ArrayList<>();
+
+        triviaQuestionSetList = triviaApiDataList.stream()
                 .map(triviaApiData -> TriviaQuestionSet.builder()
-                        .category(triviaApiData.getCategory())
-                        .type(triviaApiData.getType())
-                        .difficulty(triviaApiData.getDifficulty())
                         .question(triviaApiData.getQuestion())
                         .correct_answer(triviaApiData.getCorrect_answer())
-                        .incorrect_answers(triviaApiData.getIncorrect_answers())
+                        .answers(determineAnswers(triviaApiData))
```

Comment:
```suggestion
                        .answers(determineAnswers(triviaApiData))
                        .id(triviaApiDataList.indexOf(triviaApiData)+1)
```
i would try to set the id directly and remove addIDtoQuestionSet(triviaQuestionSetList);